File: java/ultimate/SecurityChecksMiddlewareHandler.java
package ultimate;

import chainofresponsibility.AbstractMiddlewareHandler;

public class SecurityChecksMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public MiddlewareResponse handle(HttpRequest request) {
        if (request.getUser().isHasPassedSecurityChecks()) {
            super.handle(request);
        }
        return new MiddlewareResponse("Request has failed to pass security checks", false);
    }
}

File: typescript/patterns/ultimate-design-pattern-challenge/security-checks-middleware-handler.ts
class SecurityChecksMiddlewareHandler extends AbstractMiddlewareHandler {
  override handle(request: HttpRequest): MiddlewareResponse {
    if (!request.getUser().getHasPassedSecurityChecks()) {
      return new MiddlewareResponse(
        "Request has failed to pass security checks",
        false,
      );
    }

    return super.handle(request);
  }
}
